eva_submission/eload_validation.py

```diff
@@ -353,6 +358,72 @@ def _collect_validation_workflow_results(self, output_dir):
             })
         self.eload_cfg.set('validation', 'assembly_check', 'pass', value=total_error == 0)
 
+    def _detect_structural_variant(self):
+        # Initializing the list of boolean values to store the status of the VCFs denoting whether it has a SV or not
+        has_sv = []
+        vcf_files = self._get_vcf_files()
+        for vcf_file in vcf_files:
+            has_sv_per_vcf = self._detect_structural_variant_from_variant_lines(vcf_file)
+            if not has_sv_per_vcf:
+                vcf_name = os.path.basename(vcf_file)
+                validator_output_file = os.path.join(self._get_dir('vcf_check'), vcf_name + '.vcf_validator.txt')
+                has_sv_per_vcf = self._detect_structural_variant_from_validator_output(validator_output_file)
+            has_sv.append(has_sv_per_vcf)
+        self.eload_cfg.set('validation', 'structural_variant_check', 'pass', value=True)
+        return has_sv
+
+    def _detect_structural_variant_from_variant_lines(self, vcf_file):
+        no_of_variant_lines_per_vcf = 0
+        has_sv_per_vcf = False
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Pages: 6, 16)
+        symbolic_allele_pattern = re.compile("^<(DEL|INS|DUP|INV|CNV|BND)")
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 17)
+        complex_rearrangements_breakend_pattern = re.compile(
+            "^[ATCGNatgcn]+\[.+:.+\[$|^[ATCGNatgcn]+\].+:.+\]$|^\].+:.+\][ATCGNatgcn]+$|^\[.+:.+\[[ATCGNatgcn]+$")
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 18)
+        complex_rearrangements_special_breakend_pattern = re.compile(
+            "^[ATGCNatgcn]+<[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*>$")
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 22)
+        single_breakend_pattern = re.compile("^\.[ATGCNatgcn]+|[ATGCNatgcn]+\.$")
+        if vcf_file.endswith('.vcf.gz'):
+            open_file = gzip.open(vcf_file, mode="rt")
+        else:
+            open_file = open(vcf_file, mode="r")
+        for file_line in open_file:
+            if file_line[0] == "#":
+                continue;
+            no_of_variant_lines_per_vcf = no_of_variant_lines_per_vcf + 1
+            if no_of_variant_lines_per_vcf > 10000:
+                break
+            extract_columns = file_line.split("\t")
+            alt_allele_column = extract_columns[4]
+            alternate_alleles = alt_allele_column.split(",")
+            for alternate_allele in alternate_alleles:
+                if re.search(symbolic_allele_pattern, alternate_allele) or \
+                        re.search(complex_rearrangements_breakend_pattern, alternate_allele) \
+                        or re.search(complex_rearrangements_special_breakend_pattern,
+                                     alternate_allele) or re.search(single_breakend_pattern, alternate_allele):
```

Comment:
Performance-wise, I imagine it's faster to "or" these within a compiled regex and then do a single match.  Something like:
```
symbolic_regex = "..."
breakend_regex = "..."
<insert other regex here>
sv_regex = re.compile(f'{symbolic_regex}|{breakend_regex}|...')
```
and then just a single `if sv_regex.match(alternate_allele)` here

Reply:
I have incorporated this change now